examples/component-webhooks/capabilities/hook.ts

```diff
@@ -86,46 +86,60 @@ When(a.Secret)
 async function sleepAndChangeStatus(secretName: string, componentName: string) {
   await sleep(10);
```

Comment:
I'd highly recommend replacing this, if this needs to wait for something to happen, put it in a retry loop and Log something if the time expires.

Reply:
As discussed synchronously, the sleep here isn't waiting for anything to happen. It is just there to pause execution of a zarf deployment to simulate a background process running (like creating ECR repositories).

The goal for this example isn't to necessarily demonstrate things you can automate with pepr, but rather to demonstrate how pepr can be used to inject custom logic into the deployment lifecycle, similar to zarf actions, with a key differentiator being that pepr modules are defined outside of a `zarf.yaml`. So we shouldn't need any retry logic because the only thing that is happening is zarf waiting for pepr to update the webhook status to something other than `Running` in the package secret. Zarf implements the polling/retry logic while waiting for pepr to update the webhook status here: https://github.com/defenseunicorns/zarf/blob/7cc12011f863d48168202a4a4e2fb60e501cb560/src/internal/cluster/zarf.go#L126

It would be good to set the [`WaitDurationSeconds`](https://github.com/defenseunicorns/zarf/blob/7cc12011f863d48168202a4a4e2fb60e501cb560/src/types/k8s.go#L72C5-L72C5) field to specify how long zarf should poll the webhook status before timing out as a best practice though.